subsys/bluetooth/host/l2cap.c

```diff
@@ -1053,18 +1053,18 @@ static void le_disconn_rsp(struct bt_l2cap *l2cap, u8_t ident,
 	struct bt_conn *conn = l2cap->chan.chan.conn;
 	struct bt_l2cap_le_chan *chan;
 	struct bt_l2cap_disconn_rsp *rsp = (void *)buf->data;
-	u16_t dcid;
+	u16_t scid;
 
 	if (buf->len < sizeof(*rsp)) {
 		BT_ERR("Too small LE disconn rsp packet size");
 		return;
 	}
 
-	dcid = sys_le16_to_cpu(rsp->dcid);
+	scid = sys_le16_to_cpu(rsp->dcid);
```

Comment:
As I said in my earlier comment, rsp->dcid according to the spec is  the channel endpoint on the device sending the response", i.e. since we're *receiving* the response (not sending it) rsp->dcid is the remote endpoint, not the local one. So I think you should be looking up `rsp->scid` and not `rsp->dcid` (the `scid` local variable name is appropriate however).

Reply:
yep, I forgot to change the to rsp->scid.